lib/3scale/backend/stats/partition_eraser_job.rb

```diff
@@ -0,0 +1,72 @@
+module ThreeScale
+  module Backend
+    module Stats
+      # TODO: from configuration
+      DELETE_BATCH_SIZE = 50
+      # Job for deleting service stats
+      # Perform actual key deletion from a key partition definition
+      class PartitionEraserJob < BackgroundJob
+        # low priority queue
+        @queue = :main
+
+        class << self
+          include StorageHelpers
+
+          def perform_logged(_enqueue_time, service_id, applications, metrics, users,
+                             from, to, offset, length, context_info = {})
+            job = DeleteJobDef.new(
+              service_id: service_id,
+              applications: applications,
+              metrics: metrics,
+              users: users,
+              from: from,
+              to: to
+            )
+
+            validate_job(job, offset, length)
+
+            stats_key_types = KeyTypesFactory.create(job)
+
+            stats_key_gen = KeyGenerator.new(stats_key_types)
+
+            stats_key_gen.keys.drop(offset).take(length).each_slice(DELETE_BATCH_SIZE) do |slice|
+              storage.del(slice)
+            end
+
+            [true, { job: job.to_hash, offset: offset, lenght: length }.to_json]
+          rescue Error => error
+            ErrorStorage.store(service_id, error, context_info)
+            [false, "#{service_id} #{error}"]
+          rescue Exception => error
+            if error.class == ArgumentError && error.message == 'invalid byte sequence in UTF-8'
```

Comment:
Shouldn't this be rescued somewhere closer to where it is actually being raised?

Reply:
This code has been copied from [ReportJob](https://github.com/3scale/apisonator/blob/master/lib/3scale/backend/transactor/report_job.rb), I think it does not apply for this type of job. :+1: